mscore/plugin/api/part.h

```diff
@@ -41,6 +63,21 @@ class Part : public Ms::PluginAPI::ScoreElement {
       int startTrack() const { return part()->startTrack(); }
       int endTrack()   const { return part()->endTrack(); }
       QString instrumentId() const { return part()->instrument()->instrumentId(); }
+      int harmonyCount() const { return part()->harmonyCount(); }
+      bool hasPitchedStaff() const { return part()->hasPitchedStaff(); }
+      bool hasTabStaff() const { return part()->hasTabStaff(); }
+      bool hasDrumStaff() const { return part()->hasDrumStaff(); }
+      int lyricCount() const { return part()->lyricCount(); }
+      int midiChannel() const { return part()->midiChannel(); }
+      int midiProgram() const { return part()->midiProgram(); }
+      QString longName() const { return part()->longName(); }
+      void setLongName(const QString& s) { part()->setLongName(s); }
```

Comment:
Unfortunately these `set*` functions cannot be used as they are: all changes made by plugins should be committed in an undoable way (using some functions performing changes via undo stack) so that user could undo anything that a plugin changes to a score. Unfortunately I don't currently know for sure what exactly can be used in this case (perhaps I could find something later) but these properties setters are still better not to be added to plugins API in their current state.

Reply:
They do actually function. 

I tested them against my test plug-in and score with current master plus my patch.

I didn't place those in the commit.

Reply:
Ah. I see what you're saying.

Though isn't this an issue with all plugin modifications?

Dale

Reply:
It shouldn't be, the whole plugins API is designed to make it possible for a user to revert all plugin-originated changes by simply undoing them in a usual way (the exception are plugins of `dock` type which should manage this manually using [`startCmd()/endCmd()`](https://dmitrio95.github.io/plugins/html/class_ms_1_1_plugin_a_p_i_1_1_score.html#a9970862e902ca3c0747c5aedf95b8407) calls). If it is not the case for some changes, feel free to submit a bug report, it should definitely work with undo stack correctly.

Reply:
Ok... 

So bottom line,

Should I cut them completely, comment them out, or leave them?

Reply:
You can probably remove these functions for now indeed and make the properties read-only, it should be safe to do so. You can also leave some comments reminding that thsse properties were writable in 2.X API.